frb_codegen/src/parser/ty.rs

```diff
@@ -192,22 +216,23 @@ impl<'a> TypeParser<'a> {
                 }
                 "Vec" => {
                     // Special-case Vec<String> as StringList
-                    if matches!(*generic, SupportedInnerType::Path(SupportedPathType { ref ident, .. }) if ident == "String")
+                    if matches!(p.generic[0], SupportedInnerType::Path(SupportedPathType { ref ident, .. }) if ident == "String")
                     {
                         Some(IrType::Delegate(IrTypeDelegate::StringList))
                     } else {
-                        self.convert_to_ir_type(*generic).map(|inner| match inner {
-                            Primitive(primitive) => {
-                                PrimitiveList(IrTypePrimitiveList { primitive })
-                            }
-                            others => GeneralList(IrTypeGeneralList {
-                                inner: Box::new(others),
-                            }),
-                        })
+                        self.convert_to_ir_type(p.generic.remove(0), is_exception)
```

Comment:
`p.generic.remove(0)` is seen a few times. I am wondering why we use this instead of sth like `p.generic.single()`